fat/fat2/transaction.go

```diff
@@ -0,0 +1,168 @@
+package fat2
+
+import (
+	"fmt"
+)
+
+// Transaction represents a fat2 transaction, which can be a normal account
+// transaction or a conversion transaction depending on the Inputs and Outputs
+type Transaction struct {
+	Input    TransactionIO   `json:"input"`
+	Outputs  []TransactionIO `json:"outputs"`
+	Metadata interface{}     `json:"metadata,omitempty"`
+}
+
+// Valid performs all validation checks and returns nil if t is a valid Transaction
+func (t *Transaction) Valid() error {
+	// Perform initial validation on all inputs and outputs in isolation
+	if err := t.Input.Valid(); err != nil {
+		return fmt.Errorf("invalid input: %v", err)
+	}
+	if t.Input.Address == nil {
+		return fmt.Errorf("invalid input: no address")
+	} else if t.Input.Address.String() == "FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC" {
```

Comment:
Sounds good. I had originally looked if I could find a constant somewhere, but couldn't locate it. Perhaps because I was looking for "burn" related constants